nextcord/ext/commands/converter.py

```diff
@@ -863,6 +864,74 @@ async def convert(self, ctx: Context, argument: str) -> nextcord.GuildSticker:
         return result
 
 
+_EVENT_INVITE_RE = re.compile(
+    r'(?:https?\:\/\/)?discord(?:\.gg|(?:app)?\.com\/invite)\/(.+)'
+    '?event=(\d+)'
+)
+
+_EVENT_API_RE = re.compile(
+    r'(?:https?\:\/\/)?(?:(ptb|canary|www)\.)?discord'
+    r'(?:(?:app)?\.com\/events)\/(\d+)\/(\d+)'
+)
+
+
+class ScheduledEventConverter(IDConverter[nextcord.ScheduledEvent]):
+    """Converts to a :class:`~nextcord.ScheduledEvent`.
+
+    All lookups are done for the local guild first, if available. If that lookup
+    fails, then it checks the client's global cache.
+
+    The lookup strategy is as follows (in order):
+
+    1. Lookup by ID.
+    3. Lookup by name
+    3. Lookup by url (invite?event=id and /guildid/eventid)
+
+    .. versionadded:: 2.0
+    """
+
+    async def convert(
+        self, ctx: Context, argument: str
+    ) -> nextcord.ScheduledEvent:
+        match = self._get_id_match(argument)
+        result = None
+        bot = ctx.bot
+        guild = ctx.guild
+
+        if match is None:
+            # Try to get the scheduled event by name. Try local guild first.
+            if guild:
+                result = nextcord.utils.get(guild.scheduled_events, name=argument)
+
+            if result is None:
```

Comment:
This should be removed as event names are not unique.

Reply:
eh other converters like guild use names

Reply:
True